evil-commands.el

```diff
@@ -636,6 +636,27 @@ and jump to the corresponding one."
           (when ov
             (goto-char (overlay-start ov))))))))
 
+(defun evil-ispell-mark-word-as-good (word)
+  "Add WORD under the cursor as a good word to the Ispell dictionary."
+  (interactive (list (thing-at-point 'word)))
+  (ispell-send-string (concat "*" word "\n"))
+  ;; dictionary modified!
+  (setq ispell-pdict-modified-p '(t))
+  (ispell-pdict-save ispell-silently-savep)
+  (when-let ((flyspell-mode)
```

Comment:
Don't use `when-let`. For this particular case, `(bound-and-true-p flyspell-mode)` is better. The other case is a combination of `let` and `when`.